pkg/kubectl/cmd/log.go

```diff
@@ -51,39 +51,28 @@ $ kubectl logs --since=1h nginx`
 )
 
 type LogsOptions struct {
-	Client *client.Client
-
-	PodNamespace  string
-	PodName       string
-	ContainerName string
-	Follow        bool
-	Timestamps    bool
-	Previous      bool
-	LimitBytes    int
-	Tail          int
-	SinceTime     *unversioned.Time
-	SinceSeconds  time.Duration
-
-	Out io.Writer
+	Namespace string
+	Name      string
+	Options   runtime.Object
```

Comment:
Yes, we can plug other log option objects except PodLogOptions and use them in the factory.